pulpcore/cli/common/generic.py

```diff
@@ -195,9 +210,46 @@ def get_command(self, ctx: click.Context, cmd_name: str) -> Optional[click.Comma
         if isinstance(cmd, (PulpCommand, PulpGroup)):
             if cmd.allowed_with_contexts is not None:
                 if not isinstance(ctx.obj, cmd.allowed_with_contexts):
-                    return None
+                    raise IncompatibleContext(
+                        "The command is not available under the current context."
+                    )
         return cmd
 
+    def format_commands(self, ctx: click.Context, formatter: click.HelpFormatter) -> None:
```

Comment:
Now this is really unfortunate. And right now it's hard to see which part was copied from click and what you changed to make it work. I would really want to avoid doing this. The more we monkeypatch click, the more likely it will break us in the future. We may need to think about another angle here.
What if instead we overwrite `list_commands` too to be ctx sensitive?

Reply:
I can take a look at `list_commands`.